### PR TITLE
Improve Python 3 support

### LINKED ISSUE #36: Improving Python3 support
Hi,

I've been using this plugin for a couple of years now and I just love it!

Recently I switched back to Archlinux as my main development system, also for ROS.

Unfortunately, I cannot get this plugin running in my current neovim setup, neither with the python2, nor with the python3 backend. It might be worthwhile to mention, that my ROS melodic setup on Archlinux uses python3 at most of the places...

This is not supposed to be a "Why does this not work?" issue, I would like to contribute to this myself. However, I've never done any vim-plugin development so far, any help for getting started / debugging would be nice. I didn't manage to get over the point of the following output:

```
Error detected while processing function <SNR>78_Detect[1]..provider#python3#Call:
line   18:
Error invoking 'python_execute' on channel 3 (python3-script-host):
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'vim' is not defined
Error detected while processing function <SNR>78_Detect[2]..provider#python3#Call:
line   18:
Error invoking 'python_execute' on channel 3 (python3-script-host):
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'package' is not defined
```

For testing purposes I removed all other plugins from my config, so I am pretty sure that the error comes from this plugin, but I can't figure out where to start looking. First of all, I have no idea, what this ` <SNR>78_Detect[1]..provider#python3#Call:` means and second of all I have no idea where that `line 18` references to. `name 'vim' not defined` doesn't sound too good, though.

Looking at my `pynvim` module I  can only find a name `Nvim` which seems to have the functions available used inside `ros.vim`. My assumption is that, [this line](https://github.com/taketwo/vim-ros/blob/py3/plugin/ros.vim#L94) is causing the start of trouble. If I change it to 

```
exec g:_rpy "from pynvim import Nvim as vim\npackage = rospkg.get_package_name(vim.eval('a:filename'))"
```

I get `TypeError: eval() missing 1 required positional argument: 'string'`. So seems I hit the right line and my assumption about the `Nvim` module was not completely wrong, but I think stabbing around like this won't get me too far without having an idea what I am actually doing. If you have any hints where I could start, that would be very nice :-)

I think, python3 support would be really beneficial, especially with noetic round the corner... 

I have the following versions running:
 - Neovim v0.4.3
 - pynvim 0.4.1 (python3 version)

### PR DESCRIPTION
**Pull request by fmauch**
Track changes in debugging #36 

Fixes #36

### CODE REVIEW

File: plugin/vimp/functions.py [hunk 1]
```diff
@@ -30,8 +31,8 @@ def function(name=None):
 
     def decorator(f):
         assert callable(f)
-        is_function = hasattr(f, "func_name")
-        function_name = name or (f.func_name if is_function else f.__name__)
+        is_function = inspect.isclass(f)
+        function_name = f.__name__
```
Comment (taketwo) [inline plugin/vimp/functions.py:35]:
```suggestion
         function_name = name or f.__name__ 
```

File: plugin/vimp/functions.py [hunk 2]
```diff
@@ -30,8 +31,8 @@ def function(name=None):
 
     def decorator(f):
         assert callable(f)
-        is_function = hasattr(f, "func_name")
-        function_name = name or (f.func_name if is_function else f.__name__)
+        is_function = inspect.isclass(f)
```
Comment (taketwo) [inline plugin/vimp/functions.py:34]:
```suggestion
        is_function = not inspect.isclass(f)
```

**Comment by fmauch** [comment]
I've been testing every plugin function that I know of and everything worked as expected.

Thanks again for your great help

**Comment by taketwo** [comment]
Thanks for testing and the PR.

When I have some free time, I'll test whether `py3` branch is compatible with Python 2 and, if so, merge it back into master.

**Pull request merged**